### ISSUE TITLE
pyarrow deprecated path

### ISSUE THREAD

**Issue by mfouesneau** [vaexio/vaex#174]
https://github.com/vaexio/vaex/blob/f654e895e48a1c5080ad9f84df9a2c31c7add64a/packages/vaex-arrow/vaex_arrow/dataset.py#L26

Needs to be replaced by  `pa.ipc.open_stream(source)`

**Comment by maartenbreddels** [comment]
Which (py)arrow do you have installed?

**Comment by mfouesneau** [comment]
version 0.12.0.


Also the files vaex write have issues:

```python
import pyarrow
pyarrow.open_file('data.vaex.arrow') 

ArrowInvalid: Not an Arrow file
```

It only works if read as a stream, which is not the default in other libs like dask or pandas.

**Comment by xhochy** [comment]
Note that there are two Arrow file formats:

 * The "stream" one: metadata is stored at the end of the file.
 * The "file" one: metadata is stored at the beginning of the file.

This may explain the failure in your last snippet.

**Comment by mfouesneau** [comment]
But why one instead of the other? Would be useful to have a common version
between standard packages? That's the goal of using parquet

**Comment by xhochy** [comment]
Arrow files are mainly there for temporary storage when you want to store the data uncompressed form that can be loaded directly back into memory. The "file" format is better suited for reading as you can read in the first bytes what size and type of data is coming. The "stream" format is used for the case where you write the data to disk but are not yet aware of the total amount of rows that will be written.

In contrast to this Parquet files (don't confuse them with Arrow files!) store the data in a compressed form that can be easily queried and exchanged between a lot of engines. Due to their layout, you cannot mmap Parquet files and thus they are not really suitable for `vaex`.

A good differentiator: The natural habitat of Parquet is on disk, whereas for Arrow it is in-memory.

**Comment by mfouesneau** [comment]
Ok so what's the goal of using arrow with vaex? Especially by default the
stream format. I am quite confused because if I understand the arrow format
is not good for the memory consumption. It seems a gain only for smallish
datasets.

**Comment by xhochy** [comment]
You can use Arrow to memory map data from disk to RAM. For vaex, you could either use the stream or the file format. The only change would be how the metadata is loaded. Once that is loaded you should be able to memory map both formats into RAM.

Arrow is better for memory consumption than the structures that are used for Pandas for some data types (e.g. strings) and equally as good for others. It should be equally well suited as all other backing structures that Vaex supports. This should also work for large datasets. @mfouesneau How did you come to the conclusion that it only works for smallish datasets? I would like to update some wording if this lead to confusion.

**Comment by mfouesneau** [comment]
I came to that conclusion from what you said above. If the metadata are at the end, you must read the file until the end to find the data structure. My understanding was that Arrow was a memory dataset format, but that the disk storage was parquet and not something intermediate.

Pandas and Dask etc are using also pyarrow but impose by default a File storage. My understanding was that this format is a parquet storage format. 

Therefore, I imagined that finally, one could use these files to communicate between various codes and libraries. (not just python too)
